src/lib/mina_base/zkapp_command.ml

```diff
@@ -1515,6 +1515,14 @@ struct
 
   let forget (t : t) : T.t = t.zkapp_command
 
+  (* Ensures that there's a verification_key available for all account_updates
+   * and creates a valid command associating the correct keys with each
+   * account_id.
+   *
+   * If an account_update replaces the verification_key (or deletes it),
+   * subsequent account_updates use the replaced key instead of looking in the
+   * ledger for the key (ie set by a previous transaction).
+   *)
```

Comment:
Added this here because it took me a bit to really think through all the edge cases.

Reply:
This works if there are multiple vk updates, and the most-recently-set vk is used for each update?